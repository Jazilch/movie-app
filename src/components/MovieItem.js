'use es6';

import React from 'react';

const MovieItem = ({ title, overview }) => {
  return (
    <div className="movie-item">
      <h1>{title}</h1>
      <p>{overview}</p>
    </div>
  );
};

export default MovieItem;
