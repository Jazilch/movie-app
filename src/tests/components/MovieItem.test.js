/* eslint-disable no-undef */

'use es6';

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MovieItem from '../../components/MovieItem';

describe('MovieItem component', () => {
  let movieitem;
  const title = 'Joker';
  const overview = 'Description of the Joker';

  beforeEach(() => {
    movieitem = shallow(<MovieItem title={title} overview={overview} />);
  });

  it('correctly renders movie item', () => {
    expect(toJson(movieitem)).toMatchSnapshot();
  });

  it('renders the correct text', () => {
    expect(movieitem.find('h1').text()).toEqual('Joker');
    expect(movieitem.find('p').text()).toEqual('Description of the Joker');
  });
});
