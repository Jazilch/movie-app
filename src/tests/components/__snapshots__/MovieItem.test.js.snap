// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem component correctly renders movie item 1`] = `
<MovieItem__MovieItemWrapper>
  <img
    alt="movie"
    src="http://image.tmdb.org/t/p/w342undefined"
  />
  <h1>
    Joker
  </h1>
  <p>
    Description of the Joker
  </p>
  <span>
    Crime, Drama, Thriller
  </span>
</MovieItem__MovieItemWrapper>
`;
